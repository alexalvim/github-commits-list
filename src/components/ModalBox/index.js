import React from 'react';
import { IoIosClose } from 'react-icons/io';

import ContentBox from '../ContentBox';
import {
  OverlayWrapper,
  CloseButton,
  contentBoxWrapperStyle
} from './styles';

export default ({ title, isOpen, closeModal, children }) => (
  isOpen &&
    <OverlayWrapper>
      <ContentBox
        wrapperStyles={contentBoxWrapperStyle}
        rightContent={
          <CloseButton onClick={closeModal}>
            <IoIosClose size={32}/>
          </CloseButton>
        }
        title={title}>
          {children}
      </ContentBox>
    </OverlayWrapper>
);