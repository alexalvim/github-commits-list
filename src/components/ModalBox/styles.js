import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

const OverlayWrapper = styled.div`
  align-items: center;
  background-color: ${Colors.overlayColor}
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
`;

const contentBoxWrapperStyle = `
  display: flex;
  flex-direction: column;
  height: 100%;

  @media (min-width: ${Spaces.mobileBreakpoint}) {
    max-height: 550px;
    max-width: 450px;
  }
`

export {
  OverlayWrapper,
  contentBoxWrapperStyle
}