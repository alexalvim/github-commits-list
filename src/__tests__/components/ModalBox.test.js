import React from 'react';
import { shallow } from 'enzyme';

import ModalBox from '../../components/ModalBox';

describe('Testing ModalBox', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ModalBox title='test' isOpen={true} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with children', () => {
    const wrapper = shallow(
      <ModalBox title='test' isOpen={true}>
        <div id='children'>Children</div>
      </ModalBox>
    );

    expect(wrapper.find('#children').length).toEqual(1);
  })
});