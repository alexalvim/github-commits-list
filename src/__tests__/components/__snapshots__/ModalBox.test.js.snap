// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ModalBox should render correctly 1`] = `ShallowWrapper {}`;
